### ISSUE TITLE
Conditionally render "Admin" link in navbar

### ISSUE THREAD

**Issue by ForrestLonganecker** [codeforpdx/recordexpungPDX#749]
This issue is done when:
* We keep track of user admin status in the front end.
* Render the `Admin` link when `isAdmin = true`.
* Upon an `unauthorized` response from an admin protected endpoint the variable is set back to `isAdmin = false`.

See discussion in #731

**Comment by ARodriguezHacks** [comment]
cc @JuanCaicedo

**Comment by JuanCaicedo** [comment]
@ForrestLonganecker a couple questions for you 😀

1. @ARodriguezHacks and I noticed that the `Search` nav bar link leads to an error if  you are not authorized (which we took to mean if the user is not an admin?). Should we hide that link for non-admins as well?

2. This ticket only mentions hiding the link, but we were thinking that the admin route would still be accessible if you type it directly into the url. Should we try to fix that? If so, here's my thinking on how we could do it
  - Create a function that will check if a user `isAdmin` and allow us to redirect them if they are not
  - Apply this function at the router level so that page views will get automatically rerouted

**Comment by ForrestLonganecker** [comment]
Thanks for reaching out @JuanCaicedo & @ARodriguezHacks!

1. The `Search` link is watching for the `OECI` authentication so it is currently acting the way that it should. There will be two different types of users at this point: `admin` or `search` everyone can search, but only `admins` can create new users or edit existing users. Thank you for clarifying this behavior!

2. Adding a redirect at the router level is a good idea, however there is another security concern that we will need to address at the page level as well. A react savvy person could use the react developer tools and flip the `isAdmin` flag to `true` in the dev-tools. having the redirect at the router level would be nice but we will also need to have something in the "protected" components or pages themselves that catches a `unauthorized` response from the backend and flips the `isAdmin` flag back to false/ redirect to a more appropriate page.

The most secure/seamless option would be including an endpoint that specifically returns a response stating admin/not-admin upon attempting to navigate to a protected page but at this point we are trying to limit the number of requests we are sending to the backend.

Long story short, if you would like to include a redirect at the router level based on `isAdmin` flag that would be a good improvement for now! At some point in the future we will need to implement something that will flip the flag back to false upon an unauthorized request/response.

If you have any additional questions please reach out, I really appreciate you two looking into this issue!!

**Comment by ForrestLonganecker** [comment]
There is also a little more information/discussion in issue #731
